Filter bugs

awisema opened the pull request:
Remove CoC selector.
Allow all project types.
Configuration bug was masking off project types in effective project ids.

awisema committed:
Filter bugs
Remove CoC selector.
Allow all project types.
Configuration bug was masking off project types in effective project ids.

eanders commented on drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/view_configuration.rb:
```diff
@@ -24,7 +24,7 @@ def report_path_array
 
       def default_project_type_codes
-        [:ph, :oph, :rrh, :psh]
+        GrdaWarehouse::Hud::Project::PERFORMANCE_REPORTING.keys
```
This is probably fine, but we'll need to adjust when we hit the 2024 updates.  Might be worth a TODO

awisema committed:
Note required 2022 -> 2024 change

awisema committed:
Store XLSX in S3
XLSX generation was intermittently timing out, so do it as part of report generation, and then store the file in S3.

eanders commented on drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/sheet_generator.rb:
```diff
@@ -131,6 +131,19 @@ def sheets
       end
+
+      def attach_rendered_xlsx
+        result_file.attach(filename: "Report #{id}.xlsx", io: StringIO.new(render_xlsx))
```
```suggestion
        result_file.attach(filename: "All Neighbors System Dashboard #{id}.xlsx", io: StringIO.new(render_xlsx))
```

eanders reviewed (approved):
Just a more specific name, and I think this looks good.  Thanks!

awisema committed:
Put report name in S3 filename.

awisema committed:
Merge branch 'release-84' into aw/ansd-filters-182575428

awisema committed:
Pass format to render separately to avoid deprecation warning.
HMIS-WAREHOUSE-VK
